smslibrary/src/main/java/com/eis/communication/Message.java

```diff
@@ -4,21 +4,40 @@
 
 /**
  * Encapsulation of a message in the network
+ *
  * @param <D> Data to be transmitted
  * @param <P> Peer type of users of the network
  */
-public interface  Message<D, P extends Peer> extends Serializable {
+public interface Message<D, P extends Peer> extends Serializable {
 
     /**
      * Retrieves the data sent or to be sent in the network
+     *
      * @return data contained in this message
      */
     D getData();
 
     /**
      * Retrieves the sender or the destination
+     *
      * @return Peer associated with this message
      */
     P getPeer();
 
+    /**
+     * Indicates whether some other object is "equal to" this one.
+     *
+     * @param o the reference object with which to compare.
+     * @return {@code true} if this object is the same as the o argument; {@code false} otherwise.
+     */
+    boolean equals(Object o);
+
+    /**
+     * Returns a hash code value for the object. This method is supported for the benefit of hash
+     * tables such as those provided by {@link java.util.HashMap}.
+     *
+     * @return a hash code value for this object.
+     */
+    int hashCode();
+
```

Comment:
these methods are always implemented cause all objects exteds Object
https://stackoverflow.com/questions/1718112/tostring-equals-and-hashcode-in-an-interface

Reply:
Yes, but if you don't want stuff like the memory address of the object to be included in the comparison, you need to overwrite them. When we compare two Messages we only want to know if the Peer and the content of the message are equal.

Reply:
in the link i posted they suggest to chenge the names
cause if i imlement you interface and don't write those methods no error is thrown because extending object you already implement them

Reply:
but is always a forcing

Reply:
I could make `Message` an abstract class and declare `hashCode` and `equals` `abstract`, this way classes extending `Message` would be forced to implement them

Reply:
it has the same effect as the interface

Reply:
It doesn't, see [here](https://i.imgur.com/MpgThtF.png)
Anyway, now that I think about it, the only case I can think of where a class that extends/implements `Message` needs to provide method `equals()` is when there are listeners for messages sent and messages delivered. So I probably shouldn't have added those methods to `Message` in the first place. Would it be ok if I removed them?

Reply:
sorry it may works cause you override it with a "blank page"
is the worst thing i've ever seen but may it works
this is a big problem 
"forcing to override a method foisted from java class structure" and abstract class is the only way to go
if you can remove it is better
if you need it we need to find a solution

Reply:
> sorry it may works cause you override it with a "blank page"

No, I hadn't overridden it, I was just showing you that making `Message` an abstract class would force subclasses to implement its methods. The error appeared because I had removed `equals()` from `SMSMessage`, I wanted to show you that the compiler was forcing me to implement `equals()`.
Anyway I'll remove those methods from `Message`, because they might not be always needed.